File: tests/fixtures/typescript/slop_code_uniformity.ts
export function normalizeAccount(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("account is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeProfile(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("profile is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeSession(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("session is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeInvoice(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("invoice is empty");
  }
  return value.split(" ").join("-");
}

export function normalizePayment(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("payment is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeAddress(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("address is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeContact(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("contact is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeChannel(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("channel is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeMessage(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("message is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeSummary(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("summary is empty");
  }
  return value.split(" ").join("-");
}

export function normalizeReceipt(raw: string): string {
  const value = raw.trim().toLowerCase();
  if (value.length === 0) {
    throw new Error("receipt is empty");
  }
  return value.split(" ").join("-");
}
// expect-line: 1 SLOP045

File: tests/fixtures/rust/slop_code_uniformity.rs
fn normalize_account(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-account");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_profile(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-profile");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_session(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-session");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_invoice(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-invoice");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_payment(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-payment");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_address(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-address");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_contact(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-contact");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_channel(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-channel");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_message(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-message");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_summary(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-summary");
    }
    value.replace(char::is_whitespace, "-")
}

fn normalize_receipt(raw: &str) -> String {
    let value = raw.trim().to_lowercase();
    if value.is_empty() || value == "-" {
        return String::from("unnamed-receipt");
    }
    value.replace(char::is_whitespace, "-")
}
// expect-line: 1 SLOP045
